src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -860,6 +861,20 @@ public void ReadArchive_WithUnexpectedZip64ExtraFieldSizeUncompressedSizeIn32Bit
             Assert.Equal(6, entry.CompressedLength); // it should have used 32-bit size
         }
 
+        /// <summary>
+        /// This test checks behavior of ZipArchive when the startDiskNumber in the extraField is greater than IntMax
+        /// </summary>
+        [Fact]
+        public void ReadArchive_WithDiskStartNumberGreaterThanIntMax()
+        {
+            byte[] input = (byte[])s_zip64WithBigStartDiskNumber.Clone();
+            using var archive = new ZipArchive(new MemoryStream(input));
+
+            var exception = Record.Exception(() => archive.Entries.First());
+
+            Assert.Null(exception);
```

Comment:
```suggestion
archive.Entries.First(); // does not throw

```
?

Reply:
Is there a way to validate you got the right value, not just that it didn't throw?

Reply:
Unfortunately, startDiskNumber is a private field in ZipArchiveEntry. Maybe a public property StartDiskNumber may help in this case, but i'm not sure this is a good idea to do it just for test